packages/eslint-plugin/rules/__tests__/valid-sprintf.js

```diff
@@ -37,76 +37,83 @@ ruleTester.run( 'valid-sprintf', rule, {
 		{
 			code: `var value = ''; sprintf( value, 'substitute' )`,
 		},
+		{
+			code: `
+sprintf(
+	/* translators: 1: number of blocks. 2: average rating. */
+	_n(
+		'This author has %1$d block, with an average rating of %2$d.',
+		'This author has %1$d blocks, with an average rating of %2$d.',
+		authorBlockCount
+	),
+	authorBlockCount,
+	authorBlockRating
+);`,
+		},
+		{
+			code: `i18n.sprintf( '%s', 'substitute' )`,
+		},
+		{
+			code: `i18n.sprintf( i18n.__( '%s' ), 'substitute' )`,
+		},
+		{
+			code: `sprintf( ...args )`,
+		},
 	],
 	invalid: [
 		{
 			code: `sprintf()`,
-			errors: [
-				{ message: 'sprintf must be called with a format string' },
-			],
+			errors: [ { messageId: 'noFormatString' } ],
 		},
 		{
 			code: `sprintf( '%s' )`,
-			errors: [
-				{
-					message:
-						'sprintf must be called with placeholder value argument(s)',
-				},
-			],
+			errors: [ { messageId: 'noPlaceholderArgs' } ],
 		},
 		{
 			code: `sprintf( 1, 'substitute' )`,
-			errors: [
-				{
-					message:
-						'sprintf must be called with a valid format string',
-				},
-			],
+			errors: [ { messageId: 'invalidFormatString' } ],
 		},
 		{
 			code: `sprintf( [], 'substitute' )`,
-			errors: [
-				{
-					message:
-						'sprintf must be called with a valid format string',
-				},
-			],
+			errors: [ { messageId: 'invalidFormatString' } ],
 		},
 		{
 			code: `sprintf( '%%', 'substitute' )`,
-			errors: [
-				{
-					message:
-						'sprintf format string must contain at least one placeholder',
-				},
-			],
+			errors: [ { messageId: 'noPlaceholders' } ],
 		},
 		{
 			code: `sprintf( __( '%%' ), 'substitute' )`,
-			errors: [
-				{
-					message:
-						'sprintf format string must contain at least one placeholder',
-				},
-			],
+			errors: [ { messageId: 'noPlaceholders' } ],
 		},
 		{
 			code: `sprintf( _n( '%s', '' ), 'substitute' )`,
-			errors: [
-				{
-					message:
-						'sprintf format string options must have the same number of placeholders',
-				},
-			],
+			errors: [ { messageId: 'placeholderMismatch' } ],
 		},
 		{
 			code: `sprintf( _n( '%s', '%s %s' ), 'substitute' )`,
-			errors: [
-				{
-					message:
-						'sprintf format string options must have the same number of placeholders',
-				},
-			],
+			errors: [ { messageId: 'placeholderMismatch' } ],
```

Comment:
The `messageId`s certainly make the tests a fair bit cleaner 👍 (and I assume easier to maintain if the message were ever to change)